### PR TITLE
Update all necessary follow buttons on click

### PR DESCRIPTION
**Pull request by AshWhitear**
- Added loop for all .follow-action-button
- Added check for current button dataset.info.id against event button dataset.info.id
- Moved assignState() call into the loop, called pending id matches
- Wrapped with a couple of catches - mainly for JSON object parsing issues.

## What type of PR is this? (check all applicable)
- [ ] Refactor
- [x] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Added to documentation?
  - [ ] docs.dev.to
  - [ ] readme
  - [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?
alt-text

### CODE REVIEW

**Comment by benhalpern** [comment]
Oh cool, I really like this.

@jessleenyc or @Zhao-Andy wanna examine this and give a review?

File: app/assets/javascripts/initializers/initializeAllFollowButts.js [hunk 1]
```diff
@@ -94,7 +94,26 @@ function handleOptimisticButtRender(butt) {
...
+    try {
+        //lets try grab the event buttons info data attribute user id
+        let evFabUserId = JSON.parse(butt.dataset.info).id;
+        //now for all follow action buttons
+        document.querySelectorAll('.follow-action-button').forEach(function(fab){
+            try{
+                //lets check they have info data attributes
+                if( fab.dataset.info ){
+                    //and attempt to parse those, to grab that buttons info user id
+                    let fabUserId = JSON.parse(fab.dataset.infos).id;
```
Comment (maestromac) [inline app/assets/javascripts/initializers/initializeAllFollowButts.js]: `.infos` is a typo. It should be `.info` right?

File: app/assets/javascripts/initializers/initializeAllFollowButts.js [hunk 2]
```diff
@@ -94,7 +94,26 @@ function handleOptimisticButtRender(butt) {
...
+                    //now does that user id match our event buttons user id?
+                    if ( fabUserId && fabUserId === evFabUserId ) {
+                        //yes - time to assign the same state!
+                        assignState(fab, butt.dataset.verb);
```
Comment (maestromac) [inline app/assets/javascripts/initializers/initializeAllFollowButts.js]: Assuming there are two follow buttons for the identical user on the page. If the first follow button is clicked, `butt.dataset.verb` get modified. ie the verb change from "follow" to "unfollow", which would produce no changes on the second button. `butt.dataset.verb` should be stored in a variable before it is used as a comparison.

File: app/assets/javascripts/initializers/initializeAllFollowButts.js (hunk 1 above)
Comment (AshWhitear) [reply app/assets/javascripts/initializers/initializeAllFollowButts.js]: Ooops, yes sorry you're spot on - I think I left my test var in there 😅

File: app/assets/javascripts/initializers/initializeAllFollowButts.js (hunk 2 above)
Comment (AshWhitear) [reply app/assets/javascripts/initializers/initializeAllFollowButts.js]: Ah yup, got you! Sorry, think I got too excited to contribute! 😅 All sorted and updated pushed.

File: app/assets/javascripts/initializers/initializeAllFollowButts.js [hunk 3]
```diff
@@ -94,7 +94,27 @@ function handleOptimisticButtRender(butt) {
     showModal('follow-button');
   } else {
     // Handles actual following of tags/users
-    assignState(butt, butt.dataset.verb);
+    try {
+        //lets try grab the event buttons info data attribute user id
+        let evFabUserId = JSON.parse(butt.dataset.info).id;
```
[the remark this answers is not in the document]
Comment (AshWhitear) [reply app/assets/javascripts/initializers/initializeAllFollowButts.js]: Ah okay got you, sorry about that i had no idea 😅 Should be all sorted and pushed now..

File: app/assets/javascripts/initializers/initializeAllFollowButts.js [hunk 4]
```diff
@@ -94,7 +94,27 @@ function handleOptimisticButtRender(butt) {
...
+        let requestVerb = butt.dataset.verb;
+        //now for all follow action buttons
+        document.querySelectorAll('.follow-action-button').forEach(function(fab){
+            try{
+                //lets check they have info data attributes
+                if( fab.dataset.info ){
+                    //and attempt to parse those, to grab that buttons info user id
+                    let fabUserId = JSON.parse(fab.dataset.info).id;
```
[the remark this answers is not in the document]
Comment (AshWhitear) [reply app/assets/javascripts/initializers/initializeAllFollowButts.js]: ✔

File: app/assets/javascripts/initializers/initializeAllFollowButts.js [hunk 5]
```diff
@@ -94,7 +94,27 @@ function handleOptimisticButtRender(butt) {
     showModal('follow-button');
   } else {
     // Handles actual following of tags/users
-    assignState(butt, butt.dataset.verb);
+    try {
+        //lets try grab the event buttons info data attribute user id
+        let evFabUserId = JSON.parse(butt.dataset.info).id;
+        let requestVerb = butt.dataset.verb;
```
[the remark this answers is not in the document]
Comment (AshWhitear) [reply app/assets/javascripts/initializers/initializeAllFollowButts.js]: ✔

**Comment by AshWhitear** [comment]
Thanks @Zhao-Andy, you're being too kind 😅 -and thank you all for reviewing (@benhalpern & @maestromac) and letting me contribute 😄

**Pull request merged**